src/ucs/arch/aarch64/cpu.h

```diff
@@ -38,6 +38,11 @@ BEGIN_C_DECLS
 #define ucs_aarch64_dmb(_op)          asm volatile ("dmb " #_op ::: "memory")
 #define ucs_aarch64_isb(_op)          asm volatile ("isb " #_op ::: "memory")
 #define ucs_aarch64_dsb(_op)          asm volatile ("dsb " #_op ::: "memory")
+/**
+ * Data Gathering Hint on Arm, see kernel patch:
+ * https://lore.kernel.org/linux-arm-kernel/20211221035556.60346-1-wangxiongfeng2@huawei.com/T/
+ */
+#define ucs_aarch64_dgh()             asm volatile ("hint #6" : : : "memory")
```

Comment:
does it need a special compiler?

Reply:
Nope. Since I use hint instruction with hint ID instead of actual DGH instruction, I don't need special compiler. If hint is not implemented, it is nop and if it is implemented it will execute DGH. Linux kernel is doing exactly the same thing.

Reply:
https://developer.arm.com/documentation/ddi0602/2021-12/Base-Instructions/HINT--Hint-instruction-